tests/registry.rs

```diff
@@ -1,111 +1,101 @@
+        let mut i = lock.generate_sources_for_file(&en_us, "menu.ftl");
 
-    assert_eq!(i.next(), Some(toolkit));
-    assert_eq!(i.next(), Some(browser));
-    assert_eq!(i.next(), None);
+        assert_eq!(i.next(), Some(toolkit));
+        assert_eq!(i.next(), Some(browser));
+        assert_eq!(i.next(), None);
 
-    assert!(browser.fetch_file_sync(&en_us, "menu.ftl").is_none());
+        assert!(browser.fetch_file_sync(&en_us, "menu.ftl").is_none());
 
-    let mut i = reg.generate_sources_for_file(&en_us, "menu.ftl");
-    assert_eq!(i.next(), Some(toolkit));
-    assert_eq!(i.next(), None);
+        let mut i = lock.generate_sources_for_file(&en_us, "menu.ftl");
+        assert_eq!(i.next(), Some(toolkit));
+        assert_eq!(i.next(), None);
 
-    assert!(toolkit.fetch_file_sync(&en_us, "menu.ftl").is_some());
+        assert!(toolkit.fetch_file_sync(&en_us, "menu.ftl").is_some());
 
-    let mut i = reg.generate_sources_for_file(&en_us, "menu.ftl");
-    assert_eq!(i.next(), Some(toolkit));
-    assert_eq!(i.next(), None);
+        let mut i = lock.generate_sources_for_file(&en_us, "menu.ftl");
+        assert_eq!(i.next(), Some(toolkit));
+        assert_eq!(i.next(), None);
+    }
 }
 
 #[test]
 fn test_generate_source_permutations() {
     let en_us: LanguageIdentifier = "en-US".parse().unwrap();
-    let fs1 = FileSource::new(
+    let fs1 = l10nregistry::tokio::file_source(
         "toolkit".to_string(),
         vec![en_us.clone()],
         "./data/toolkit/{locale}/".into(),
-        fetch_sync,
     );
-    let fs2 = FileSource::new(
+    let fs2 = l10nregistry::tokio::file_source(
         "browser".to_string(),
         vec![en_us.clone()],
         "./data/browser/{locale}/".into(),
-        fetch_sync,
     );
 
-    let mut reg = L10nRegistry::new();
-
+    let mut reg = L10nRegistry::default();
     reg.register_sources(vec![fs1, fs2]).unwrap();
 
-    let toolkit = reg.get_source("toolkit").unwrap();
-    let browser = reg.get_source("browser").unwrap();
+    {
+        // let lock = reg.lock();
+        // let toolkit = lock.get_source("toolkit").unwrap();
+        // let browser = lock.get_source("browser").unwrap();
 
-    let paths = vec!["menu.ftl", "brand.ftl"];
-    let mut i = reg.generate_source_permutations(&en_us, &paths);
+        // let paths = vec!["menu.ftl", "brand.ftl"];
+        // let mut i = lock.generate_source_permutations(&en_us, &paths);
 
-    assert_eq!(i.next(), Some(vec![toolkit, toolkit]));
-    assert_eq!(i.next(), Some(vec![toolkit, browser]));
-    assert_eq!(i.next(), Some(vec![browser, toolkit]));
-    assert_eq!(i.next(), Some(vec![browser, browser]));
-    assert_eq!(i.next(), None);
+        // assert_eq!(i.next(), Some(vec![toolkit, toolkit]));
+        // assert_eq!(i.next(), Some(vec![toolkit, browser]));
+        // assert_eq!(i.next(), Some(vec![browser, toolkit]));
+        // assert_eq!(i.next(), Some(vec![browser, browser]));
+        // assert_eq!(i.next(), None);
```

Comment:
leftover?

Reply:
I removed the `source_permutations` API since it wasn't used except for testing.